src/GitVersionExe/Program.cs

```diff
@@ -129,17 +129,27 @@ static void ConfigureLogging(Arguments arguments)
             {
                 try
                 {
-                    Directory.CreateDirectory(Path.GetDirectoryName(arguments.LogFilePath));
-                    if (File.Exists(arguments.LogFilePath))
+                    var logFileFullPath = Path.GetFullPath(arguments.LogFilePath);
+                    var logFile = new FileInfo(logFileFullPath);
+
+                    if (logFile.Directory == null)
+                        throw new DirectoryNotFoundException(String.Format("The directory of {0} does not exist.", logFile));
+
+                    if (!logFile.Directory.Exists)
```

Comment:
I would remove `logFile.Directory.Exists` here since it's not needed and you cannot be sure if the directory is there in the next statement.